common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -258,7 +257,7 @@ class WebSocketRepositoryImpl @Inject constructor(
 
     private suspend fun connect(): Boolean {
         connectedMutex.withLock {
-            if (connection != null && connected.isCompleted) {
+            if (connection != null && connected.isCompleted && connected.await()) {
```

Comment:
Just making sure this is intentional: because `connected.await()` returns the result, if it is `false` this block will be skipped. With this change the app will immediately continue trying to set up a new connection, instead of the current behavior where it returns and the connection has to be set to null first (in `handleClosingSocket`). Is that the intended effect?